tasks/libs/ciproviders/gitlab_api.py

```diff
@@ -84,25 +114,150 @@ def __init__(self, references):
 
     @classmethod
     def from_yaml(cls, loader, node):
-        return UserList(loader.construct_sequence(node))
+        return YamlReferenceTagList(loader.construct_sequence(node))
 
     @classmethod
     def to_yaml(cls, dumper, data):
-        return dumper.represent_sequence(cls.yaml_tag, data.data, flow_style=True)
+        return dumper.represent_sequence(cls.yaml_tag, data, flow_style=True)
+
+
+def convert_to_config_node(node):
+    """
+    Convert json data to ConfigNode
+    """
+    if isinstance(node, dict):
+        return ConfigNodeDict({k: convert_to_config_node(v) for k, v in node.items()})
+    elif isinstance(node, list):
+        constructor = YamlReferenceTagList if isinstance(node, YamlReferenceTagList) else ConfigNodeList
+
+        return constructor([convert_to_config_node(v) for v in node])
```

Comment:
```suggestion
        return ConfigNodeList([convert_to_config_node(v) for v in node])
```
If my comment above is correct this is simpler like this, wdyt?